spyder/app/tests/conftest.py

```diff
@@ -325,12 +325,13 @@ def main_window(request, tmpdir, qtbot):
     use_introspection = request.node.get_closest_marker('use_introspection')
 
     if use_introspection:
-        os.environ['SPY_TEST_USE_INTROSPECTION'] = 'True'
+        CONF.set('completions', ('enabled_providers', 'lsp'), True)
+        CONF.set('completions', ('enabled_providers', 'fallback'), True)
+        CONF.set('completions', ('enabled_providers', 'snippets'), True)
     else:
-        try:
-            os.environ.pop('SPY_TEST_USE_INTROSPECTION')
-        except KeyError:
-            pass
+        CONF.set('completions', ('enabled_providers', 'lsp'), False)
+        CONF.set('completions', ('enabled_providers', 'fallback'), False)
+        CONF.set('completions', ('enabled_providers', 'snippets'), False)
```

Comment:
Just checking ... it is not necessary to undo these changes in the CONF after the test, is it?

Reply:
I don't think so but to be sure checked and seems like the logic that enables not worrying about possible effects to the config after running a test is at: 

https://github.com/spyder-ide/spyder/blob/071e646e26d005b30fb838dd215b5a5ca4312ffb/spyder/config/base.py#L219-L224 

So when running tests, a clean dir is used for the config :+1: